hazelcast/src/test/java/com/hazelcast/client/protocol/compatibility/ClientCompatibilityTest_2_0.java

```diff
@@ -6917,32 +6920,24 @@ private void compareClientMessages(ClientMessage binaryMessage, ClientMessage en
 
         ClientMessage.ForwardFrameIterator binaryFrameIterator = binaryMessage.frameIterator();
         ClientMessage.ForwardFrameIterator encodedFrameIterator = encodedMessage.frameIterator();
+        assertTrue("Client message that is read from the binary file does not have any frames", binaryFrameIterator.hasNext());
 
-        boolean isInitialFramesCompared = false;
         while (binaryFrameIterator.hasNext()) {
             binaryFrame = binaryFrameIterator.next();
             encodedFrame = encodedFrameIterator.next();
             assertNotNull("Encoded client message has less frames.", encodedFrame);
 
-            boolean isFinal = binaryFrameIterator.peekNext() == null;
-            if (!isInitialFramesCompared) {
-                compareInitialFrame(binaryFrame, encodedFrame, isFinal);
-                isInitialFramesCompared = true;
-            } else {
-                assertArrayEquals("Frames have different contents", binaryFrame.content, encodedFrame.content);
-                int flags = isFinal ? encodedFrame.flags | IS_FINAL_FLAG : encodedFrame.flags;
-                assertEquals("Frames have different flags", binaryFrame.flags, flags);
+            if (binaryFrame.isEndFrame() && !encodedFrame.isEndFrame()) {
+                if (encodedFrame.isBeginFrame()) {
+                    HazelcastClientUtil.fastForwardToEndFrame(encodedFrameIterator);
+                }
+                encodedFrame = HazelcastClientUtil.fastForwardToEndFrame(encodedFrameIterator);
```

Comment:
What is the reason of having to calls to `fastForwardToEndFrame` ? Why do we need to fast forward once more if `encodedFrame.isBeginFrame()`

Reply:
Lets say Address(a custom type) is extended with a new field : an AttributeConfig(another custom type). When we do the comparison, when we are at the end frame of the binary message, we will be at the begin frame of the AttributeConfig field of the encoded message. If I do just one fast forward at this point, I will be fast forwarding to the END_FRAME of the AttributeConfig not the encoded message. For this special case, we fast forward twice.